xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/internal/UntypedEventListener.java

```diff
@@ -0,0 +1,191 @@
+import org.xwiki.rendering.renderer.BlockRenderer;
+import org.xwiki.rendering.renderer.printer.DefaultWikiPrinter;
+import org.xwiki.rendering.renderer.printer.WikiPrinter;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateManager;
+
+/**
+ * This listener listens to every event occurring on the wiki. When an event happens, it tries to determine if this
+ * event should trigger another one (an {@link UntypedRecordableEvent} which is described by a
+ * {@link UntypedRecordableEventDescriptor}) given the current context.
+ *
+ * @version $Id$
+ * @since 9.6RC1
+ */
+@Component
+@Singleton
+@Named(UntypedEventListener.NAME)
+public class UntypedEventListener extends AbstractEventListener
+{
+    /**
+     * The listener name.
+     */
+    public static final String NAME = "Untyped Event Listener";
+
+    /**
+     * The binding name of the event when a validation expression is rendered.
+     */
+    public static final String EVENT_BINDING_NAME = "event";
+
+    @Inject
+    private ObservationManager observationManager;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    @Named("plain/1.0")
+    private BlockRenderer plainRenderer;
+
+    @Inject
+    @Named("wiki")
+    private ComponentManager componentManager;
+
+    @Inject
+    private ModelBridge modelBridge;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Constructs a {@link UntypedEventListener}.
+     */
+    public UntypedEventListener()
+    {
+        super(NAME, AllEvent.ALLEVENT);
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        try {
+            // Get every UntypedEventDescriptor registered in the ComponentManager
+            List<UntypedRecordableEventDescriptor> descriptors =
+                    componentManager.getInstanceList(UntypedRecordableEventDescriptor.class);
+
+            // Filter the event descriptors concerned by the event, then create the concerned events
+            for (UntypedRecordableEventDescriptor descriptor : descriptors) {
+                // If the event is expected by our descriptor
+                if (descriptor.getEventTriggers().contains(event.getClass().getCanonicalName())
+                        && checkXObjectCondition(descriptor, source)
+                        && this.evaluateVelocityTemplate(
+                        event, descriptor.getAuthorReference(), descriptor.getValidationExpression()))
+                {
+                    observationManager.notify(
+                            new DefaultUntypedRecordableEvent(descriptor.getEventType()),
+                            "org.xwiki.platform:xwiki-platform-eventstream-api",
+                            source);
+                }
+            }
+        } catch (ComponentLookupException e) {
+            logger.warn(
```

Comment:
We don't display the stack trace when it's a warning. But we use ExceptionUtilse#getRootMessage since e.getMessage() is usually close to useless.

That being said this looks like a serious issue and should IMO be logged as an error with the complete stack trace instead.